### PR TITLE
47 section background colour

### LINKED ISSUE #47: FR: Section metadata options: background colour
By default, sections (or page fragments) don’t have a background colour defined. On the [PDP](https://www.figma.com/file/yv0VtHTvMoGPqkv0RAYcVp/PDP?type=design&node-id=1091-23111&mode=dev) and [in modals on the PDP](https://www.figma.com/file/yv0VtHTvMoGPqkv0RAYcVp/PDP?type=design&node-id=1741-24323&mode=dev), we can have a dark or [grey background](https://www.figma.com/file/yv0VtHTvMoGPqkv0RAYcVp/PDP?type=design&node-id=3781-47970&mode=dev).

### Requirements
- A section can have a modifier class "grey" or "black"
- Utilising CSS Custom Properties, colours of backgrounds and text will change in components
  - Rename `--c-background-card` to `--card-background`
  - Add `--card-text`
- Tertiary / text buttons will follow the [Core Design Kit](https://www.figma.com/file/JglsLWtLIgdFru0p8Va2vz/BETA---Volvo-Core-Design-Kit?type=design&node-id=535-22166&mode=design&t=7Whg1qSDrRsD8cHz-4), and will always be shown in uppercase.

### PR DESCRIPTION
**Pull request by avinashNChavan**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #47 

Test URLs:
- Before: https://main--vg-volvotrucks-us-rd--netcentric.hlx.page/
- After: https://47-section-background-colour--vg-volvotrucks-us-rd--netcentric.hlx.page/drafts/avinash/V2/background

### CODE REVIEW

File: styles/styles.css [hunk 1]
```diff
@@ -636,6 +634,16 @@ main .section.highlight {
   background-color: var(--highlight-background-color);
 }
 
+.section-dark {
```
[the remark this answers is not in the document]
Comment (kesiah) [reply styles/styles.css:637]: yes, please use `section--grey`, so we make sure we are using modifiers

[the remark this answers is not in the document]
Comment (kesiah) [reply styles/styles.css:637]: Also, I'm using `dark` class in the section to change the colour of the text in Hero and Banner blocks, could we use `black` instead for the black background (#141414) and grey for the other one (#F7F7F7)?

[the remark this answers is not in the document]
Comment (avinashNChavan) [reply styles/styles.css]: Updated

File: styles/styles.css [hunk 2]
```diff
@@ -636,6 +634,16 @@ main .section.highlight {
...
+  background-color: var(--c-main-black);
+  color: var(--c-white);
+}
+
+.section-darkgrey {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply styles/styles.css]: Updated

File: scripts/scripts.js [hunk 3]
```diff
@@ -333,11 +334,17 @@ function buildTruckCarouselBlock(main) {
...
-  main.querySelectorAll(':scope > .section[data-background]').forEach((section) => {
-    const src = section.dataset.background;
-    const picture = createOptimizedPicture(src, '', false);
-    section.appendChild(picture);
-    section.classList.add('section-with-background');
+  main.querySelectorAll(':scope > .section').forEach((section) => {
+    const variantClasses = ['black', 'grey'];
+    variantsClassesToBEM(section.classList, variantClasses, 'section');
+
+    // console.log(main.querySelectorAll(':scope > .section[data-background]'));
```
Comment (TomaszDziezykNetcentric) [inline scripts/scripts.js:341]: please remove commented code

**Review by TomaszDziezykNetcentric — CHANGES_REQUESTED** [review]

File: scripts/scripts.js [hunk 4]
```diff
@@ -333,11 +334,17 @@ function buildTruckCarouselBlock(main) {
...
+    if (section.dataset.background) {
+      const src = section.dataset.background;
+      const picture = createOptimizedPicture(src, '', false);
```
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply scripts/scripts.js:344]: This is existing function which is being modified to accommodate the new background color changes. There are places where background images is used in current site, so we shouldn't remove that class

File: styles/styles.css [hunk 5]
```diff
@@ -636,6 +636,16 @@ main .section.highlight {
   background-color: var(--highlight-background-color);
 }
 
+.section--black {
+  background-color: var(--c-main-black);
+  color: var(--c-white);
+}
+
+.section--grey {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply styles/styles.css:645]: Updated

File: styles/styles.css [hunk 6]
```diff
@@ -1123,8 +1133,10 @@ main.blue-contract .section.section-with-title p {
 
 .redesign-v2 a.button.tertiary,
 .redesign-v2 button.tertiary {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply styles/styles.css:1136]: Updated

File: blocks/v2-dealer-locator-teaser/v2-dealer-locator-teaser.css [hunk 7]
```diff
@@ -54,6 +54,11 @@
   margin: 0;
 }
 
+.v2-dlt__content .icon svg {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-dealer-locator-teaser/v2-dealer-locator-teaser.css]: Updated

File: styles/styles.css [hunk 8]
```diff
@@ -636,6 +637,16 @@ main .section.highlight {
   background-color: var(--highlight-background-color);
 }
 
+.section--black {
+  background-color: var(--c-main-black);
+  color: var(--c-white);
+}
+
+.section--grey {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply styles/styles.css]: Updated

File: styles/styles.css [hunk 9]
```diff
@@ -168,8 +168,9 @@
   --input-background-color: var(--c-grey-1);
   --overlay-background-color: var(--c-grey-1);
   --light-border-focus: #2B8EDE;
-  --card-background: var(--c-grey-50);
-  --card-text: var(--ff-volvo-novum);
+  --card-background: var(--c-grey-900);
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply styles/styles.css]: Updated

**Pull request closed without merging**